src/librbd/internal.cc

```diff
@@ -1597,11 +1620,6 @@ int invoke_async_request(ImageCtx *ictx, const std::string& request_type,
         }
 
         disable_flags = RBD_FLAG_OBJECT_MAP_INVALID;
-        r = remove_object_map(ictx);
```

Comment:
I would keep this here as well.  If we crash between removing the object map and clearing the feature bit, the image will still function w/ object map implicitly disabled.  If we move this down, if we crash between clearing the bit and remove the object map, we leave a object behind.